model/anomaly_nn.ipynb

```diff
@@ -1,991 +1,1322 @@
-    "        \n",
-    "    model.eval()\n",
-    "    with torch.set_grad_enabled(False): # save memory during inference\n",
-    "        train_performance = compute_f1(model, train_loader, device=DEVICE)\n",
-    "        val_performance = compute_f1(model, val_loader, device=DEVICE)\n",
-    "        epoch_train_performance.append(train_performance) # from lecture 3 example files\n",
-    "        epoch_val_performance.append(val_performance)\n",
-    "        print('Epoch: %03d/%03d | Train: %.3f%%   | Val: %.3f%%' % (\n",
-    "              epoch+1, NUM_EPOCHS, train_performance, val_performance))\n",
-    "        \n",
-    "    print('Time elapsed: %.2f min' % ((time.time() - start_time)/60))\n",
-    "    \n",
-    "print('Total Training Time: %.2f min' % ((time.time() - start_time)/60))"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 412,
-   "metadata": {},
-   "outputs": [
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "ta7khuA6F3e7",
+        "outputId": "d86b51af-eac2-48ae-f33d-2b07ba176356"
+      },
+      "source": [
+        "train_labels = train_labels.reset_index(drop=True)\n",
+        "val_labels = val_labels.reset_index(drop=True)\n",
+        "\n",
+        "print(\"train data shape\")\n",
+        "print(train_data.shape)\n",
+        "print(train_labels.shape)\n",
+        "print(\"val data shape\")\n",
+        "print(val_data.shape)\n",
+        "print(val_labels.shape)"
+      ],
+      "execution_count": 7,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "train data shape\n",
+            "(315602, 16, 46)\n",
+            "(315602,)\n",
+            "val data shape\n",
+            "(78901, 16, 46)\n",
+            "(78901,)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "sPraFUqn-yUx"
+      },
+      "source": [
+        "class logDataset(Dataset):\n",
+        "    \"\"\"Log Anomaly Features Dataset\"\"\"\n",
+        "    \n",
+        "    def __init__(self, data_vec, labels=None):\n",
+        "        self.X = data_vec\n",
+        "        self.y = labels\n",
+        "        \n",
+        "    def __len__(self):\n",
+        "        return len(self.X)\n",
+        "        \n",
+        "    def __getitem__(self, idx):\n",
+        "        data_matrix = self.X[idx]\n",
+        "        \n",
+        "        if not self.y is None:\n",
+        "            return(data_matrix, self.y[idx])\n",
+        "        else:\n",
+        "            return data_matrix\n"
+      ],
+      "execution_count": 9,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Ol2-sxy5-yUy"
+      },
+      "source": [
+        "# Convert to tensort and pad matrixes to have dims of 20x48 instead of 19x48\n",
```

Comment:
Is this still true for our dimensions? Should the padding be removed here - note that if it is removed we may need to modify some dimensions of the model (can always figure this out using print statements)